src/App.jsx

```diff
@@ -10,14 +10,25 @@ import {
 import { bioData } from './assets/bioData.js';
 import { Home, Layout, ProfessionalsPage } from './views';
 
-function App() {
+
   // SET STATES
-  const [data, setData] = useState(bioData);
+  
+  function App() {
+    const [data, setData] = useState(bioData);
+    
+  const getCards = (tag) =>{
+   const newData =bioData.filter((ele)=> {
```

Comment:
After reviewing all the code, I think we should use the `data` state instead of `bioData` in this function. Whether or not `data` is updated from `bioData` when this function is called, using data state will allow us to move this function definition into the component where the function is invoked, leading to more code readability and separation of concerns.

Reply:
@Havana82 made the great point that: using `data` instead of `bioData` wouldn’t work because `data` changes every time you call `setData` on it and it won’t be filtering the original list after the first click

Reply:
the `useRef` I set up in `App.jsx` comes in handy here! `fullDataSet` is already passed as a prop to `ProfessionalsPage` - from there you can pass it to your component and use that to filter!